src/NuGet.Clients/NuGet.PackageManagement.UI/Telemetry/ActionsTelemetryExtensions.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Telemetry;
+using NuGet.PackageManagement.Telemetry;
+using NuGet.PackageManagement.UI;
+using NuGet.Versioning;
+using NuGet.VisualStudio.Internal.Contracts;
+using TelemetryEvent = NuGet.Common.TelemetryEvent;
+
+namespace NuGet.VisualStudio
+{
+    internal static class ActionsTelemetryExtensions
+    {
+        internal static void AddUiActionEngineTelemetryProperties(
```

Comment:
If line count is a concern for getting this in, you can consider reverting these refactorings.

Reply:
Ok. Reverting in a bit.

Reply:
Reverted: